agent/xds/clusters.go

```diff
@@ -1865,8 +1868,8 @@ func (s *ResourceGenerator) makeExternalIPCluster(snap *proxycfg.ConfigSnapshot,
 }
 
 // makeExternalHostnameCluster creates an Envoy cluster for hostname endpoints that will be resolved with DNS
-// This is used by both terminating gateways for Destinations, and Mesh Gateways for peering control plane traffice
-func (s *ResourceGenerator) makeExternalHostnameCluster(snap *proxycfg.ConfigSnapshot, opts clusterOpts) *envoy_cluster_v3.Cluster {
+// This is used by both terminating gateways for Destinations, and Mesh Gateways for peering control plane traffic
+func (s *ResourceGenerator) makeExternalHostnameCluster(snap *proxycfg.ConfigSnapshot, opts clusterOpts, discoveryType envoy_cluster_v3.Cluster_DiscoveryType) *envoy_cluster_v3.Cluster {
```

Comment:
Does it make sense for discoveryType to be added to clusterOpts instead of as a separate entry field? Otherwise looks good to me, thanks for knocking this out.

Reply:
I don't think so, mainly b/c `clusterOps` is used a bunch of places in terms of cluster generation and most existing functions would never use it -- I don't like the idea of having a "catch all" options structure where only a couple of functions actually use certain fields on the structure, it makes usage harder to track down IMO.